inst/pages/98_exercises.qmd

```diff
@@ -1149,12 +1440,137 @@ Useful functions: scater::runMDS, runNMDS, transformAssay, ggplot, scater::plotR
+#| eval: false
+
+tse <- transformAssay(tse, assay.type = "counts", method = "relabundance")
+```
+
+4. Visualize the relative abundances with a heatmap using the sechm library
+
+```{r}
+#| label: heatmap-ex4
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+setSechmOption("hmcols", value=c("#F0F0FF","#007562"))
+
+sechm(tse,features = most_prevalent , assayName="relabundance",        show_colnames=TRUE)
+```
+
+### Advanced Heatmaps
+
+1. Import mia, miaviz, complexHeatmap, shadowtext as well as a dataset of your choice
+
+```{r}
+#| label: advanced-heatmap-ex1
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+library(mia)
+library(miaViz)
+library(shadowtext)
+library(ComplexHeatmap) 
+
+data("Tengeler2020", package = "mia")
+tse <- Tengeler2020
+```
+
+2. Agglomerate the data by prevalence to a rank of your choice, apply log10 transformation and assure unique rownames for when we plot the heatmap.
+
+```{r}
+#| label: advanced-heatmap-ex2
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+tse <- agglomerateByPrevalence(tse, rank = "Family")
+
+tse <- transformAssay(tse, method = "log10", pseudocount = TRUE)
+
+rownames(tse) <- getTaxonomyLabels(tse)
+
+```
+
+3. Calculate the correlations and their respective p-values between the log0 and count assays.
+
+```{r}
+#| label: advanced-heatmap-ex3
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+correlations <- getCrossAssociation(tse, 
+                                   test_significance=TRUE,            
+                                   assay.type1 = "log10", 
+                                   assay.type2 = "counts",
+                                   method = "spearman", 
+                                   p_adj_threshold = NULL,
+                                   cor_threshold = NULL,
+                                   # Remove when mia is fixed
+                                   mode = "matrix",
+                                   sort = TRUE,
+                                   show_warnings = FALSE)
+```
+
+4. plot a heatmap using the correlations and mark significant correlations (p<0.05)with a cross.
+```{r}
+#| label: advanced-heatmap-ex4
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+# Create a heatmap and store it
+plot <- Heatmap(correlations$cor,
+                # Print values to cells
+                cell_fun = function(j, i, x, y, width, height, fill) {
+                  # If the p-value is under threshold
+                  if( !is.na(correlations$pval[i, j]) &                                        correlations$pval[i, j] < 0.05 ){
+                    # Print "X"
+                    grid.shadowtext(sprintf("%s", "X"), x, y, gp =                           gpar(fontsize = 8, col = "#f5f5f5"))
+                  }
+                },
+                heatmap_legend_param = list(title = "", legend_height = unit(5, "cm"))
+)
+plot
+```
+
+
+=======
```

Comment:
this should be removed?

Reply:
Also there is no reply for this (or modification as suggested)